apps/src/code-studio/components/ExportDialog.jsx

```diff
@@ -206,25 +219,95 @@ class ExportDialog extends React.Component {
   };
 
   componentDidUpdate(prevProps) {
-    const {isOpen, projectUpdatedAt} = this.props;
-    const {exportProjectUpdatedAt} = this.state;
+    const {isOpen, md5SavedSources} = this.props;
+    const {md5PublishSavedSources} = this.state;
     if (isOpen && !prevProps.isOpen) {
       recordExport('open');
-      if (projectUpdatedAt !== exportProjectUpdatedAt) {
-        // The project has changed since we last opened the dialog,
-        // reset our export state, so we will need to export again:
-        this.resetExportState();
+      const md5ApkSavedSources = this.getMd5ApkSavedSources();
+      const sourcesChangedSinceExport =
+        md5ApkSavedSources && md5SavedSources !== md5ApkSavedSources;
+      const sourcesChangedThisInstance =
+        md5PublishSavedSources && md5SavedSources !== md5PublishSavedSources;
+      if (sourcesChangedSinceExport) {
```

Comment:
It's not clear why we would want to cancel the build when the page reloads. Wouldn't we want it to continue?

Reply:
We always assume that is the sources have changed and you have entered into this dialog, all older builds should be canceled. This allows you to actually do an export with your current source.

Reply:
👍 for this PR. I think this fits in with the discussion below. Hopefully we can rethink this as part of that discussion.